rootfs/etc/nginx/lua/metric_emitter.lua

```diff
@@ -0,0 +1,50 @@
+local socket = ngx.socket.udp
+local cjson = require('cjson')
+local assert = assert
+
+local _M = {
+  queue = {}
```

Comment:
`queue` can be local, no? why does it have to be exported?

Reply:
This is not needed anymore since we are going to use the `defer` module to delegate deferring callbacks (with queue optimization) to the timer phase.

Reply:
@fmejia97 could you use this version of the module instead: https://github.com/Shopify/ingress/blob/master/rootfs/etc/nginx/lua/util/defer.lua? 

This doesn't export things that don't need to be exported